docs/learn/learn-xcm-pallet.md

```diff
@@ -10,12 +10,90 @@ slug: ../learn-xcm-pallet
+local or external chains. `pallet-xcm`'s functionality is separated into three categories:
+
+:::note
+
+Remember, all XCM messages are effectively XCVM programs that contain a set of instructions. It is
+the job of the XCM exector to handle these programs, and their subsequent instructions as needed.
+
+:::
 
 1. Primitive, dispatchable functions to locally execute an XCM message.
 2. High-level, dispatchable functions for asset transfers.
 3. Version negotiation-specific dispatchable functions.
+
+## Primitive Extrinsics
+
+There are two primary primitive extrinsics. These extrinsics handle sending and executing XCVM
+programs as dispatchable functions within the pallet.
+
+1. `execute` - This call is direct access to the XCM executor. It checks the origin and message and
+   ensures that no barrier/filter will block the execution of the XCM message. Once it is deemed
+   valid, the message will then be _locally_ executed, therein returning the outcome as an event.
+   This operation is executed on behalf of whichever account has signed the extrinsic. It's possible
+   for only a partial execution to occur.
+2. `send` - This call specifies where a message should be sent externally to a particular
+   destination, i.e., another parachain. It checks the origin, destination, and message and is then
+   sent to the `XcmRouter`.
+
+:::info
+
+The XCM pallet needs the `XcmRouter` to send messages. It is used to dictate where messages are
+allowed to be sent, and which XCM transport protocol to use. For example, Kusama, the canary
+network, uses the `ChildParachainRoute` which only allows for Downward Message Passing from the
+relay to parachains to occur.
+
+You can read more about [XCM transport methods here.](./learn-xcm-transport.md)
+
+:::
+
+## Asset Transfer Extrinsics
+
+Several extrinsics within the pallet handle asset transfer logic. They define a predetermined set of
+instructions for sending and executing XCM messages. Two variants of these functions are prefixed
+with `limited_`. They have the same functionality but can specify a weight to pay for the XCM fee.
+
+Otherwise, the fee is taken as needed from the asset being transferred.
+
+1. `reserve_transfer_assets` - Transfer some assets from the local chain to the sovereign account of
+   a destination chain and forward a notification XCM.
+2. `teleport_assets` - Teleport some assets from the local chain to some destination chain.
+
+### Transfer Reserve vs Teleport
+
+While both extrinsics deal with transferring assets, they exhibit fundamentally different behavior.
+
+- **Teleporting** an asset implies a two-step process: the asset is burned/destroyed in the origin
+  chain and re-minted to whatever account is specified at the destination. Teleporting should only
+  occur if there is an inherent and bilateral trust between the two chains, as the tokens destroyed
+  _could not_ necessarily be guaranteed to have the same properties when minted at the destination.
```

Comment:
This isn't the fundamental problem -- the more fundamental problem is whether or not the two chains can trust each other to have burnt or minted the funds beforehand. A chain can simply send the `ReceiveTeleportedAsset` XCM instruction without previously having burnt any.